### PR TITLE
grab fix + physics logging/debug changes

### PR DESCRIPTION
**Pull request by sethalves**
- keep viewers in sync with someone using AvatarActionHold
- change how physics and networking visual debugging is handled
- attempt to fix rendering of collision hulls (this doesn't work yet, will continue in another PR)

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 774e2ab6d6338f6c6d95426ae6c9c0c3f3616288 available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

File: libraries/entities/src/EntityScriptingInterface.cpp [hunk 1]
```diff
@@ -611,7 +611,8 @@ QUuid EntityScriptingInterface::addAction(const QString& actionTypeString,
                                           const QVariantMap& arguments) {
     QUuid actionID = QUuid::createUuid();
     auto actionFactory = DependencyManager::get<EntityActionFactoryInterface>();
-    bool success = actionWorker(entityID, [&](EntitySimulation* simulation, EntityItemPointer entity) {
+    bool success = false;
```
Comment (ZappoMan) [inline libraries/entities/src/EntityScriptingInterface.cpp]: what's going on here... this looks fishy.

File: libraries/entities/src/EntityScriptingInterface.cpp [hunk 2]
```diff
@@ -611,7 +611,8 @@ QUuid EntityScriptingInterface::addAction(const QString& actionTypeString,
...
+    actionWorker(entityID, [&](EntitySimulation* simulation, EntityItemPointer entity) {
```
Comment (ZappoMan) [inline libraries/entities/src/EntityScriptingInterface.cpp]: why have actionWorker return bool if you're not using it

Comment (sethalves) [inline libraries/entities/src/EntityScriptingInterface.cpp]: it returns what the thunk returned, simply to make some calls to it cleaner.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 66387c45307db73c8ef8aee357e72fb95a8a0588 available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4cf84a4e5b940bc7ac61d3f3d8ed615a9e1590b9 available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b38c1cb30f9fce0f9fe24430ceadb6390697af7c available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c2b0ccd6b395ec6a5729adfc17f0cac15f98a4d1 available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 31ebe5dba1f210203705e249b4e01f1296fadf6c available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f07d5d9d3f290f161171d33f659e3d02ee4ffb6e available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: cffe4551ddc47fbb36d433970d783ad5d5b0e5fe available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d1a9a32e91c6a20c37c0dd803ca224048b68190f available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: be8674820022dca2efdeaa82fd0addcefe722919 available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a16a477134380d32e10d7dd289dc488278b140e3 available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 83d5c9cf4a367e871dd3445033e23ac15ee32133 available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3519dd248b734d71d828c1a6151c76adc3eefd6f available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a366633f843e45178ceeae1f3cf74bc3c4fae2a7 available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2e635bd93638413fe9d66d14d7bb0bd576f1a85b available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a8cd891e48f6b2a53aeeedfd6c9f34ee66a086d1 available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a78b1619be9d71cfa6a6f0016aa4dd26dec3dfcc available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d7be1699a7f0d5e13e735029f2b9996daf69194f available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6f83f32eccbc2726e9153c3bd3ea8f2e2d19b2ab available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 756b382a75981593971073257de5d89233e0219c available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ed8a604e136dd257f090d34428cd7b92d04da7a5 available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: dbaea5770aba054360001819fbbd454f4242c93d available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7979fb38cfa526bcd300f7b37726f5ffdc76a6fd available at: https://gist.github.com/b4fd43f90fda4909aaa6<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6ec5b12081bd52b1ba602e824d5d1918208fc27e available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3f7f5a2bbd7c3015b0fe38e4a296d69e0112e56c available at: https://gist.github.com/1c0ad97015ef4edd260b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5acbb7436262cd5abefcb07be85c1d3d158fbda7 available at: https://gist.github.com/7d320ade1f9fdb138ca7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6ec5b12081bd52b1ba602e824d5d1918208fc27e available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1d0a2ddd3d80cdbb17b1427c291461837a37a082 available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 919ea5f1a05f42645db8678f79af83cd9e264058 available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 37469206552b12e270cdabf5a8323f227e33edae available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3b6b56f316574fd8d41c02e05e6b92c6d133705e available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg

File: libraries/entities-renderer/src/RenderableModelEntityItem.cpp [hunk 3]
```diff
@@ -209,11 +192,11 @@ bool RenderableModelEntityItem::addToScene(EntityItemPointer self, std::shared_p
     
     if (_model) {
         render::Item::Status::Getters statusGetters;
-        makeEntityItemStatusGetters(this, statusGetters);
+        makeEntityItemStatusGetters(shared_from_this(), statusGetters);
```
Comment (Atlante45) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: Nice catch

File: libraries/render/src/render/DrawStatus.cpp [hunk 4]
```diff
@@ -29,7 +29,7 @@ using namespace render;
 
 
 
-const gpu::PipelinePointer& DrawStatus::getDrawItemBoundsPipeline() {
+const gpu::PipelinePointer DrawStatus::getDrawItemBoundsPipeline() {
```
Comment (Atlante45) [inline libraries/render/src/render/DrawStatus.cpp]: Why did you remove the **&**?

File: libraries/render/src/render/DrawStatus.cpp [hunk 5]
```diff
@@ -112,11 +126,31 @@ void DrawStatus::run(const SceneContextPointer& sceneContext, const RenderContex
                     (*itemAABox).setBox(item.bounds.getCorner(), 0.1f);
                 }
                 auto& itemScene = scene->getItem(item.id);
-                (*itemStatus) = itemScene.getStatusPackedValues();
+
+                auto itemStatusPointer = itemScene.getStatus();
+                if (itemStatusPointer) {
+                    // Query the current status values, this is where the statusGetter lambda get called
+                    auto&& currentStatusValues = itemStatusPointer->getCurrentValues();
```
Comment (Atlante45) [inline libraries/render/src/render/DrawStatus.cpp]: No need for **&&**, **&** is enough

Comment (sethalves) [inline libraries/render/src/render/DrawStatus.cpp]: compiler does not agree:

DrawStatus.cpp:133:85: error: invalid initialization of non-const reference of type ‘std::vector<render::Item::Status::Value>&’ from an rvalue of type ‘render::Item::Status::Values {aka std::vector<render::Item::Status::Value>}’

File: libraries/render/src/render/DrawStatus.cpp (hunk 4 above)
Comment (sethalves) [inline libraries/render/src/render/DrawStatus.cpp]: Sam did this -- I think he did it because it's already a smartpointer type?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 89ed3751ae3b306909e377c74bf64e718de26998 available at: https://gist.github.com/927ad3cbbaa79f601252<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6365.dmg
